### PR TITLE
Read the manifest.yml into a structure data

### PR DESCRIPTION
**Pull request by showpune**

### CODE REVIEW

File: azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/command/ResourceCommand.java [hunk 1]
```diff
@@ -26,6 +26,7 @@ public void execute(String commandText) {
         final String resources;
         try {
             resources = resourceFacade.listResource();
+            resourceFacade.initApplicationConfiguration();
```
Comment (qiaoleiatms) [inline azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/command/ResourceCommand.java:29]: why initAppConfig here? what if user just give a folder only contains cf manifests file without source code? should we let user directly go the final step to generate bicep?

File: azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/service/resource/ApplicationConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.victools.jsonschema.generator.*;
+import com.github.victools.jsonschema.module.jackson.JacksonModule;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+@JsonClassDescription("The Application configuration")
+public class ApplicationConfiguration {
```
Comment (qiaoleiatms) [inline azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/service/resource/ApplicationConfiguration.java:19]: cf manifest file is starts with 
```yaml
---
applications:
- name: xxxx
```
it's an array object, how should we deal with multiple apps scenario?

File: azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/command/ResourceCommand.java (hunk 1 above)
Comment (showpune) [reply azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/command/ResourceCommand.java:29]: What I am thinking is to prefullfill some field into the variables, so we can skip some set steps. For example
1) In manifest.yml, the db connection is setted, so we can skip this step
2) In manifest.yml, the env is already setted, we can keep it

File: azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/service/resource/ApplicationConfiguration.java (hunk 2 above)
Comment (showpune) [reply azure-java-migration-copilot-client/src/main/java/com/azure/migration/java/copilot/service/resource/ApplicationConfiguration.java:19]: Yes I thought this issue, there should be lots of work need to do, we support single one first

**Pull request merged**